internal/prcomment/github.go

```
@@ -0,0 +1,117 @@
+package prcomment
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/olbrichattila/qreview/internal/env"
+	"github.com/olbrichattila/qreview/internal/git"
+)
+
+func newGitHub(env env.EnvironmentManager) (Commenter, error) {
+	if env == nil {
+		return nil, fmt.Errorf("please provide github token in your environment: `GITHUB_TOKEN`")
+	}
+
+	return &github{
+		env: env,
+	}, nil
+}
+
+type github struct {
+	env env.EnvironmentManager
+}
+
+// Comment implements Commenter.
+func (g *github) Comment(prURL, filePath string, comment string, lineNumber int) error {
+	githubToken := g.env.GithubToken()
+	owner, repo, prNumber, err := git.GetPRInfo(prURL)
+	if err != nil {
+		return err
+	}
+
+	commitSHA, err := g.getPRHeadSHA(githubToken, owner, repo, prNumber)
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/pulls/%d/comments", owner, repo, prNumber)
+
+	if lineNumber == 0 {
+		lineNumber = 1
+	}
+
+	body := map[string]interface{}{
+		"body":      comment,
+		"commit_id": commitSHA,
+		"path":      filePath,
+		"line":      lineNumber,
+		"side":      "RIGHT",
+	}
+
+	jsonBody, err := json.Marshal(body)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonBody))
+	if err != nil {
```

The function silently ignores API errors by returning nil when the status code indicates failure. This masks errors and can make debugging difficult.